src/CachePlugin.php

```diff
@@ -0,0 +1,162 @@
+<?php
+
+namespace Http\Client\Plugin;
+
+use Psr\Cache\CacheItemInterface;
+use Psr\Cache\CacheItemPoolInterface;
+use Psr\Http\Message\RequestInterface;
+use Psr\Http\Message\ResponseInterface;
+
+/**
+ * Allow for caching a response.
+ *
+ * @author Tobias Nyholm <tobias.nyholm@gmail.com>
+ */
+class CachePlugin implements Plugin
+{
+    /**
+     * @var CacheItemPoolInterface
+     */
+    private $pool;
+
+    /**
+     * Default time to store object in cache. This value is used if CachePlugin::respectCacheHeaders is false or
+     * if cache headers are missing.
+     *
+     * @var int
+     */
+    private $defaultTtl;
+
+    /**
+     * Look at the cache headers to know how long this response is going to be cached.
+     *
+     * @var bool
+     */
+    private $respectCacheHeaders;
+
+    /**
+     * @param CacheItemPoolInterface $pool
+     * @param int                    $defaultTtl
+     * @param bool                   $respectCacheHeaders
+     */
+    public function __construct(CacheItemPoolInterface $pool, $defaultTtl = 60, $respectCacheHeaders = true)
```

Comment:
+1 for a $options array with `default_ttl` and `respect_cache_headers` options for now. will be easier to scale.